### ISSUE TITLE
training a segmentation model, but predictions are only detections

### ISSUE THREAD

**Issue by mehran66** [ultralytics/ultralytics#2974]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am training a segmentation model but during the training the (M) metrics are zero and finally during the predictions, only bounding boxes are predicted.

epoch | train/box_loss | train/seg_loss | train/cls_loss | train/dfl_loss | metrics/precision(B) | metrics/recall(B) | metrics/mAP50(B) | metrics/mAP50-95(B) | metrics/precision(M) | metrics/recall(M) | metrics/mAP50(M) | metrics/mAP50-95(M) | val/box_loss | val/seg_loss | val/cls_loss | val/dfl_loss | lr/pg0 | lr/pg1 | lr/pg2
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | 0.78758 | 0.59958 | 0.95794 | 1.1544 | 0.90713 | 0.82945 | 0.91389 | 0.79476 | 0 | 0 | 0 | 0 | 0.52692 | 0.58576 | 0.48954 | 0.9759 | 0.070046 | 0.0033282 | 0.0033282

######################################################
here is the config file:

  ```My config file is:
  # Ultralytics YOLO 🚀, GPL-3.0 license
  # Default training settings and hyperparameters for medium-augmentation COCO training
  
  task: segment  # YOLO task, i.e. detect, segment, classify, pose
  mode: train  # YOLO mode, i.e. train, val, predict, export, track, benchmark
  
  # Train settings -------------------------------------------------------------------------------------------------------
  model: yolov8x-seg.pt # path to model file, i.e. yolov8n.pt, yolov8n.yaml
  data: '/shared/data/personal_data/experiments/yolo/data/full_data_2.yaml'  # path to data file, i.e. coco128.yaml
  epochs: 150  # number of epochs to train for
  patience: 140  # epochs to wait for no observable improvement for early stopping of training
  batch: 64  # number of images per batch (-1 for AutoBatch)
  imgsz: 768  # size of input images as integer or w,h
  save: True  # save train checkpoints and predict results
  save_period: -1 # Save checkpoint every x epochs (disabled if < 1)
  cache: True  # True/ram, disk or False. Use cache for data loading
  device: 0,1,2,3,4,5,6,7  # device to run on, i.e. cuda device=0 or device=0,1,2,3 or device=cpu
  workers: 16  # number of worker threads for data loading (per RANK if DDP)
  project: /shared/data/personal_data/experiments/yolo/runs # project name
  name: ex2 # experiment name
  exist_ok: False  # whether to overwrite existing experiment
  pretrained: False  # whether to use a pretrained model
  optimizer: SGD  # optimizer to use, choices=['SGD', 'Adam', 'AdamW', 'RMSProp']
  verbose: True  # whether to print verbose output
  seed: 0  # random seed for reproducibility
  deterministic: True  # whether to enable deterministic mode
  single_cls: False  # train multi-class data as single-class
  rect: False  # rectangular training if mode='train' or rectangular validation if mode='val'
  cos_lr: True  # use cosine learning rate scheduler
  close_mosaic: 30  # (int) disable mosaic augmentation for final epochs
  resume: False  # resume training from last checkpoint
  amp: True  # Automatic Mixed Precision (AMP) training, choices=[True, False], True runs AMP check
  fraction: 1.0  # dataset fraction to train on (default is 1.0, all images in train set)
  profile: False  # profile ONNX and TensorRT speeds during training for loggers
  # Segmentation
  overlap_mask: False  # masks should overlap during training (segment train only)
  mask_ratio: 1  # mask downsample ratio (segment train only)
  # Classification
  dropout: 0.0  # use dropout regularization (classify train only)
  
  # Val/Test settings ----------------------------------------------------------------------------------------------------
  val: True  # validate/test during training
  split: val  # dataset split to use for validation, i.e. 'val', 'test' or 'train'
  save_json: False  # save results to JSON file
  save_hybrid: False  # save hybrid version of labels (labels + additional predictions)
  conf:  # object confidence threshold for detection (default 0.25 predict, 0.001 val)
  iou: 0.7  # intersection over union (IoU) threshold for NMS
  max_det: 300  # maximum number of detections per image
  half: False  # use half precision (FP16)
  dnn: False  # use OpenCV DNN for ONNX inference
  plots: True  # save plots during train/val
  
  # Prediction settings --------------------------------------------------------------------------------------------------
  source: /shared/data/personal_data/experiments/yolo/test # source directory for images or videos
  show: False  # show results if possible
  save_txt: True  # save results as .txt file
  save_conf: True  # save results with confidence scores
  save_crop: False  # save cropped images with results
  show_labels: True  # show object labels in plots
  show_conf: True  # show object confidence scores in plots
  vid_stride: 1  # video frame-rate stride
  line_thickness: 1  # bounding box thickness (pixels)
  visualize: False  # visualize model features
  augment: False  # apply image augmentation to prediction sources
  agnostic_nms: False  # class-agnostic NMS
  classes:  # filter results by class, i.e. class=0, or class=[0,2,3]
  retina_masks: False  # use high-resolution segmentation masks
  boxes: True  # Show boxes in segmentation predictions
  
  # Export settings ------------------------------------------------------------------------------------------------------
  format: torchscript  # format to export to
  keras: False  # use Keras
  optimize: False  # TorchScript: optimize for mobile
  int8: False  # CoreML/TF INT8 quantization
  dynamic: False  # ONNX/TF/TensorRT: dynamic axes
  simplify: False  # ONNX: simplify model
  opset:  # ONNX: opset version (optional)
  workspace: 4  # TensorRT: workspace size (GB)
  nms: False  # CoreML: add NMS
  
  # Hyperparameters ------------------------------------------------------------------------------------------------------
  lr0: 0.01  # initial learning rate (i.e. SGD=1E-2, Adam=1E-3)
  lrf: 0.00001  # final learning rate (lr0 * lrf)
  momentum: 0.937  # SGD momentum/Adam beta1
  weight_decay: 0.0005  # optimizer weight decay 5e-4
  warmup_epochs: 3.0  # warmup epochs (fractions ok)
  warmup_momentum: 0.8  # warmup initial momentum
  warmup_bias_lr: 0.1  # warmup initial bias lr
  box: 7.5  # box loss gain
  cls: 0.5  # cls loss gain (scale with pixels)
  dfl: 1.5  # dfl loss gain
  pose: 12.0  # pose loss gain
  kobj: 1.0  # keypoint obj loss gain
  label_smoothing: 0.0  # label smoothing (fraction)
  nbs: 64  # nominal batch size
  hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
  hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
  hsv_v: 0.4  # image HSV-Value augmentation (fraction)
  degrees: 45.0  # image rotation (+/- deg)
  translate: 0.1  # image translation (+/- fraction)
  scale: 0.5  # image scale (+/- gain)
  shear: 0.0  # image shear (+/- deg)
  perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
  flipud: 0.0  # image flip up-down (probability)
  fliplr: 0.5  # image flip left-right (probability)
  mosaic: 1.0  # image mosaic (probability)
  mixup: 0.2  # image mixup (probability)
  copy_paste: 0.0  # segment copy-paste (probability)
  
  # Custom config.yaml ---------------------------------------------------------------------------------------------------
  cfg:  # for overriding defaults.yaml
  
  # Debug, do not modify -------------------------------------------------------------------------------------------------
  v5loader: False  # use legacy YOLOv5 dataloader
  
  # Tracker settings ------------------------------------------------------------------------------------------------------
  tracker: botsort.yaml  # tracker type, ['botsort.yaml', 'bytetrack.yaml']

```

I run the training like this:

setsid nohup yolo segment train cfg=./configs/config_training_ex2.yaml > ./runs/log_ex2.out & disown

checking the data and it is all good in the segmentation format....the images plotted by the model verifies that too.
and here is the beginning of the log file that shows everything is correct for the segmentation model:

```
Overriding /home/ubuntu/.local/lib/python3.11/site-packages/ultralytics/yolo/cfg/default.yaml with ./configs/config_training_ex2.yaml
WARNING ⚠️ 'line_thickness' is deprecated and will be removed in 'ultralytics 8.2' in the future. Please use 'line_width' instead.
Ultralytics YOLOv8.0.112 🚀 Python-3.11.3 torch-2.0.1 CUDA:0 (NVIDIA A10G, 22723MiB)
                                                      CUDA:1 (NVIDIA A10G, 22723MiB)
                                                      CUDA:2 (NVIDIA A10G, 22723MiB)
                                                      CUDA:3 (NVIDIA A10G, 22723MiB)
                                                      CUDA:4 (NVIDIA A10G, 22723MiB)
                                                      CUDA:5 (NVIDIA A10G, 22723MiB)
                                                      CUDA:6 (NVIDIA A10G, 22723MiB)
                                                      CUDA:7 (NVIDIA A10G, 22723MiB)
[34m[1myolo/engine/trainer: [0mtask=segment, mode=train, model=yolov8x-seg.pt, data=/shared/data/personal_data/experiments/yolo/data/full_data_bf_2.yaml, epochs=150, patience=140, batch=64, imgsz=768, save=True, save_period=-1, cache=True, device=0,1,2,3,4,5,6,7, workers=16, project=/shared/data/personal_data/experiments/yolo/runs, name=ex2, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=30, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=False, mask_ratio=1, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=/shared/data/personal_data/experiments/yolo/test, show=False, save_txt=True, save_conf=True, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=1, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=1e-05, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=45.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.2, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=/shared/data/personal_data/experiments/yolo/runs/ex2
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1      2320  ultralytics.nn.modules.conv.Conv             [3, 80, 3, 2]                 
  1                  -1  1    115520  ultralytics.nn.modules.conv.Conv             [80, 160, 3, 2]               
  2                  -1  3    436800  ultralytics.nn.modules.block.C2f             [160, 160, 3, True]           
  3                  -1  1    461440  ultralytics.nn.modules.conv.Conv             [160, 320, 3, 2]              
  4                  -1  6   3281920  ultralytics.nn.modules.block.C2f             [320, 320, 6, True]           
  5                  -1  1   1844480  ultralytics.nn.modules.conv.Conv             [320, 640, 3, 2]              
  6                  -1  6  13117440  ultralytics.nn.modules.block.C2f             [640, 640, 6, True]           
  7                  -1  1   3687680  ultralytics.nn.modules.conv.Conv             [640, 640, 3, 2]              
  8                  -1  3   6969600  ultralytics.nn.modules.block.C2f             [640, 640, 3, True]           
  9                  -1  1   1025920  ultralytics.nn.modules.block.SPPF            [640, 640, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  3   7379200  ultralytics.nn.modules.block.C2f             [1280, 640, 3]                
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  3   1948800  ultralytics.nn.modules.block.C2f             [960, 320, 3]                 
 16                  -1  1    922240  ultralytics.nn.modules.conv.Conv             [320, 320, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  3   7174400  ultralytics.nn.modules.block.C2f             [960, 640, 3]                 
 19                  -1  1   3687680  ultralytics.nn.modules.conv.Conv             [640, 640, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  3   7379200  ultralytics.nn.modules.block.C2f             [1280, 640, 3]                
 22        [15, 18, 21]  1  12317171  ultralytics.nn.modules.head.Segment          [1, 32, 320, [320, 640, 640]] 
YOLOv8x-seg summary: 401 layers, 71751811 parameters, 71751795 gradients

Transferred 651/657 items from pretrained weights
DDP command: ['/home/ubuntu/mambaforge/envs/yolo/bin/python3.11', '-m', 'torch.distributed.run', '--nproc_per_node', '8', '--master_port', '57921', '/home/ubuntu/.config/Ultralytics/DDP/_temp_eooh4w2j139926816048656.py']
WARNING:__main__:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
DDP info: RANK 0, WORLD_SIZE 8, DEVICE cuda:0
[34m[1mTensorBoard: [0mStart with 'tensorboard --logdir /shared/data/personal_data/experiments/yolo/runs/ex2', view at http://localhost:6006/
Overriding model.yaml nc=80 with nc=1
Transferred 651/657 items from pretrained weights
[34m[1mAMP: [0mrunning Automatic Mixed Precision (AMP) checks with YOLOv8n...
[34m[1mAMP: [0mchecks passed ✅
[34m[1moptimizer:[0m SGD(lr=0.01) with parameter groups 106 weight(decay=0.0), 117 weight(decay=0.0005), 116 bias
```

**Comment by MERYX-bh** [comment]
I'm having the same problem

**Comment by glenn-jocher** [comment]
@mehran66 with YOLOv8. Can you please provide more details about the issue you are facing? Is there an error message being displayed? Have you tried any troubleshooting steps? Let me know so we can figure out a solution together.

**Comment by mehran66** [comment]
@glenn-jocher Thanks for your response.....There is not any error or warning during training, validation and prediction...I only see that all of the (M) metrics during training are zero and then the mask* plots are blank plots (just the zero line). Also all of the prediction plots only show the bounding box, though the plots for lables are all fine with both the segments and bboxes. also lokking at the txt files genrated after running inference, it is like this (only two numbers per line):
0 0.971923
0 0.966911
0 0.96664
0 0.964346
0 0.956572

The only test that I ran was training on 1 gpu rather than 8 and faced the same issue. I am not sure what other tests I can run as my data looks good and I have installed the latest version of ultralytics using pip.

**Comment by Laughing-q** [comment]
@mehran66 noticed you were using `overlap_mask=False`, I'm able to reproduce the issue with `overlap_mask=False`. Probably some updates broke it, I'll take a look.

**Comment by Laughing-q** [comment]
@mehran66 @MERYX-bh fixed in #2985

**Comment by Laughing-q** [comment]
@mehran66 I'm not able to reproduce the issue.
```python
from ultralytics import YOLO

model = YOLO("yolov8n-seg.pt")
model.train(data="coco128-seg.yaml", overlap_mask=False)
```
![image](https://github.com/ultralytics/ultralytics/assets/61612323/3f5989a3-8fc2-4339-a1b8-e9749882cbe7)

**Comment by mehran66** [comment]
@matdealm

**Comment by glenn-jocher** [comment]
@matdealm hi there, 

When running YOLOv8, it's important to run the command through Python instead of directly on the terminal. This allows the code to execute properly and produce the expected results. So, I recommend trying to run the command in a Python environment to see if that resolves the issue. Let me know if you have any further questions or need any assistance with the process.

**Comment by glenn-jocher** [comment]
@matdealm That's great to hear! I'm glad to know running it in a Python environment worked for you. If you have any more questions or run into any issues, feel free to reach out. Happy coding with YOLOv8!
